File: applications/event-account-gemfire-sink/src/main/java/showcase/event/stream/rabbitmq/account/functions/AccountConsumer.java
/*
 *
 *  * Copyright 2023 VMware, Inc.
 *  * SPDX-License-Identifier: GPL-3.0
 *
 */
package showcase.event.stream.rabbitmq.account.functions;

import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.Account;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import showcase.event.stream.rabbitmq.account.repostories.AccountRepository;

import java.util.function.Consumer;

@RequiredArgsConstructor
@Component
@Slf4j
public class AccountConsumer implements Consumer<Account> {
    private final AccountRepository repository;

    public void accept(Account account) {
        log.info("Saving account: {}",account);
        repository.save(account);
    }
}

File: applications/account-jdbc-amqp-sink/src/main/kotlin/com/vmware/tanzu/data/services/rabbitmq/streaming/account/consumer/AccountConsumer.kt
package com.vmware.tanzu.data.services.rabbitmq.streaming.account.consumer

import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.AccountEntity
import com.vmware.tanzu.data.services.rabbitmq.streaming.account.repository.AccountRepository
import org.springframework.stereotype.Component
import java.util.function.Consumer

/**
 * @author Gregory Green
 */
@Component
class AccountConsumer(private val accountRepository: AccountRepository) :
    Consumer<AccountEntity> {
    override fun accept(account: AccountEntity) {
        accountRepository.save(account)
    }
}
